File: service/src/main/java/hu/bsstudio/safetycar/SafetyCarService.java
package hu.bsstudio.safetycar;

import hu.bsstudio.safetycar.model.SafetyCarFollowInformation;
import hu.bsstudio.safetycar.model.SafetyCarOvertakeInformation;
import hu.bsstudio.team.model.DetailedTeam;
import reactor.core.publisher.Mono;

public interface SafetyCarService {
    Mono<DetailedTeam> safetyCarWasFollowed(final SafetyCarFollowInformation safetyCarFollowInformation);

    Mono<DetailedTeam> safetyCarWasOvertaken(final SafetyCarOvertakeInformation safetyCarOvertakeInformation);
}

File: service/src/main/kotlin/hu/bsstudio/robonaut/safetycar/SafetyCarService.kt
package hu.bsstudio.robonaut.safetycar

import hu.bsstudio.robonaut.safetycar.model.SafetyCarFollowInformation
import hu.bsstudio.robonaut.safetycar.model.SafetyCarOvertakeInformation
import hu.bsstudio.robonaut.team.model.DetailedTeam
import reactor.core.publisher.Mono

interface SafetyCarService {
    fun safetyCarWasFollowed(safetyCarFollowInformation: SafetyCarFollowInformation): Mono<DetailedTeam>
    fun safetyCarWasOvertaken(safetyCarOvertakeInformation: SafetyCarOvertakeInformation): Mono<DetailedTeam>
}
